package.json

```diff
@@ -10,26 +10,53 @@
     "headless cms"
   ],
   "engines": {
-    "node": ">=0.10.32"
+    "node": ">=16",
```

Comment:
Since this is an npm package, this `engines` property impacts the consuming developer's project.

What this will do to someone using buttercms is tell them they must be using Node 16 or higher, they must use pnpm and must use pnpm v7 or higher.

While this approach works for projects that are not npm packages, we might need to take a different approach here.

Since you've stated the lowest supported version of Node is 12.20.0, we should put `"node": ">=12.20.0"`, so anyone using a lower version of this is notified of deprecated support. You can still keep Node 18 as the pipeline this project runs on, but it might be best to also drop the pipeline to v12.20.0 for coverage. I'm not too sure of what the best approach would be on that latter statement, so might require some research.

---

As for the package manager, this is only an internal tool to developers of this package and would not necessarily affect the package size or end user performance. It's also tricky to enforce a standard package manager, unless you use the `packageManager` property in `package.json`.

My recommendation would be to remove `npm` and `pnpm` from `engines` and either:
- Use `packageManager` (but accept you'd be limited to specifying a specific version)
- OR stick to what the Butter team are used to => npm (in which case, it still might be worth specifying `packageManager` to ensure a standard of npm since there are huge differences between major versions)

Reply:
Good point. I've removed the `pnpm` bits and changed node to `12.20.0` which is min supported version. Also added 'todo' item to confirm preferred package manager with Butter team.

Reply:
Brill. Might also be worth researching if the pipeline should run Node 12.20.0 or Node 18 as I can see benefits to both 🤔

Reply:
There is no reason I can think of for why we'd run pipeline in Node 12, the `isomorphic-unfetch` package uses `node-fetch` - its `node-fetch` which supports down to v12. The pipeline is just compiling files and not impacting how these packages function.